user-service/src/main/java/rs/edu/raf/banka1/services/implementations/OrderServiceImpl.java

```diff
@@ -91,8 +102,69 @@ public void startOrder(final Long orderId) {
     }
 
     @Override
-    public void changeStatus(final Long id, final OrderStatus orderStatus) {
-        orderRepository.changeStatus(id, orderStatus);
+    public boolean changeStatus(final Long id, final OrderStatus orderStatus) {
+        Optional<MarketOrder> optOrder = this.orderRepository.findById(id);
+        if (optOrder.isEmpty()) return false;
+        MarketOrder order = optOrder.get();
+        order.setStatus(orderStatus);
+        order.setDone(true); // Status changed
+        order.setLastModifiedDate(System.currentTimeMillis() / 1000);
+        this.orderRepository.save(order);
+        return true;
+    }
+
+    private User getLoggedUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        // Check if the user is authenticated
+        if (authentication != null && authentication.getPrincipal() instanceof UserDetails) {
+            // Assuming your UserDetails implementation has the email field
+            UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+            String email = userDetails.getUsername();
+            //return this.agentRepository.findByEmail(email);
+            //TODO waiting for agent implementation
+        }
+
+        return null;
+    }
```

Comment:
Ovde treba da se doda?

Reply:
to jest ovaj ``agentRepository``

Reply:
Pa da, moramo da vidimo kako cemo to implementirati i sta cemo sa tim agentom pa cu dopuniti.

Reply:
Važi.